central/networkgraph/service/service_impl_postgres_benchmark_test.go

```diff
@@ -0,0 +1,302 @@
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/fixtures/fixtureconsts"
+	"github.com/stackrox/rox/pkg/networkgraph/externalsrcs"
+	"github.com/stackrox/rox/pkg/networkgraph/testutils"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/timestamp"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	globalWriteAccessCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.NetworkGraph, resources.Deployment)))
+
+	// all entity IPs of the form 1.2.x.y/32
+	baseIp = "1.2"
+)
+
+type networkGraphServiceBenchmarks struct {
+	db *pgtest.TestPostgres
+
+	service Service
+
+	clusterDataStore     clusterDS.DataStore
+	deploymentsDataStore deploymentDS.DataStore
+	entityDataStore      networkEntityDS.EntityDataStore
+	flowDataStore        networkFlowDS.ClusterDataStore
+	policyDataStore      networkPolicyDS.DataStore
+	configDataStore      configDS.DataStore
+	treeMgr              networktree.Manager
+}
+
+func setupBenchmarkForB(b *testing.B) networkGraphServiceBenchmarks {
+	var n networkGraphServiceBenchmarks
+	db := pgtest.ForT(b)
+
+	var err error
+	n.clusterDataStore, err = clusterDS.GetTestPostgresDataStore(b, db.DB)
+	require.NoError(b, err)
+
+	n.deploymentsDataStore, err = deploymentDS.GetTestPostgresDataStore(b, db.DB)
+	require.NoError(b, err)
+
+	n.entityDataStore, err = networkEntityDS.GetTestPostgresDataStore(b, db.DB)
+	require.NoError(b, err)
+
+	n.flowDataStore, err = networkFlowDS.GetTestPostgresClusterDataStore(b, db.DB)
+	require.NoError(b, err)
+
+	n.policyDataStore, err = networkPolicyDS.GetTestPostgresDataStore(b, db.DB)
+	require.NoError(b, err)
+
+	n.configDataStore, err = configDS.GetTestPostgresDataStore(b, db.DB)
+	require.NoError(b, err)
+
+	n.treeMgr = networktree.Singleton()
+
+	n.service = New(
+		n.flowDataStore,
+		n.entityDataStore,
+		n.treeMgr,
+		n.deploymentsDataStore,
+		n.clusterDataStore,
+		n.policyDataStore,
+		n.configDataStore,
+	)
+
+	n.db = db
+	return n
+}
+
+// setupTables setups up the database for a large number of deployments all
+// communicating with the same external IP - it is intended for benchmarks of the
+// GetExternalNetworkFlows API endpoint
+func (suite *networkGraphServiceBenchmarks) setupTables(b *testing.B) string {
+	cidr := "192.168.0.1/32"
+	id, err := externalsrcs.NewClusterScopedID(fixtureconsts.Cluster1, cidr)
+	require.NoError(b, err)
+
+	entity := testutils.GetExtSrcNetworkEntity(id.String(), cidr, cidr, false, fixtureconsts.Cluster1)
+
+	err = suite.entityDataStore.CreateExternalNetworkEntity(globalWriteAccessCtx, entity, true)
+	require.NoError(b, err)
+
+	flows := make([]*storage.NetworkFlow, 0, 2000)
```

Comment:
Perhaps save some of these numbers to variables.

Reply:
Have made it a parameter to this function